tower/src/retry/budget.rs

```diff
@@ -89,21 +89,51 @@ use tokio::time::Instant;
 /// For more info about [`Budget`], please see the [module-level documentation].
 ///
 /// [module-level documentation]: self
-pub struct Budget {
-    bucket: Bucket,
+pub struct Budget<B = TpsBucket> {
+    bucket: B,
     deposit_amount: isize,
     withdraw_amount: isize,
 }
 
-/// Indicates that it is not currently allowed to "withdraw" another retry
-/// from the [`Budget`].
+/// A [`Bucket`] for managing retry tokens.
+/// 
+/// [`SimpleBucket`] works by increasing the cost of retries on each withdraw.
+/// Cost multiplier decrease by 1 on each deposit.
+/// 
+/// [`Budget`] uses a token bucket to decide if the request should be retried.
+/// 
+/// For more info about [`Budget`], please see the [module-level documentation].
+/// For more info about [`Bucket`], see [bucket trait]
+/// 
+/// [module-level documentation]: self
+/// [bucket trait]: self::Bucket
 #[derive(Debug)]
-pub struct Overdrawn {
-    _inner: (),
+pub struct SimpleBucket {
```

Comment:
Instead of adding the new simple bucket here lets start smaller and just do the moving around for now. Then we can add a new impl in a future PR.